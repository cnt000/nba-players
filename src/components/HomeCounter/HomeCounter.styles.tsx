import styled from 'styled-components'

export const StyledContainer = styled.div`
  text-align: center;
`

export const StyledLogo = styled.img`
  height: 40vmin;
  pointer-events: none;
  background-color: ${({ theme }) => theme.colors.blue};
  padding: 1rem;
  border-radius: 8px;

  @media (prefers-reduced-motion: no-preference) {
    animation: App-logo-float infinite 3s ease-in-out;
  }
`

export const StyledHeader = styled.header`
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
`

export const StyledLink = styled.a`
  color: rgb(112, 76, 182);
`
