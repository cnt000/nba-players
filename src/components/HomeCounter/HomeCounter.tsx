import React from 'react'
import {
  StyledContainer,
  StyledHeader,
  StyledLogo,
  StyledLink,
} from './HomeCounter.styles'
import logo from './logo.svg'
import { Counter } from '../../state/features/counter/Counter'

const HomeCounter: React.FC = () => {
  return (
    <StyledContainer>
      <StyledHeader>
        <StyledLogo src={logo} alt="logo" />
        <Counter />
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <span>
          <span>Learn </span>
          <a
            href="https://reactjs.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            React
          </a>
          <span>, </span>
          <a
            href="https://redux.js.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Redux
          </a>
          <span>, </span>
          <StyledLink
            className="App-Styledlink"
            href="https://redux-toolkit.js.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Redux Toolkit
          </StyledLink>
          ,<span> and </span>
          <StyledLink
            className="App-link"
            href="https://react-redux.js.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            React Redux
          </StyledLink>
        </span>
      </StyledHeader>
    </StyledContainer>
  )
}

export default HomeCounter
