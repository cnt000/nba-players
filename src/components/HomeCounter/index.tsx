import HomeCounter from './HomeCounter'

export default HomeCounter
