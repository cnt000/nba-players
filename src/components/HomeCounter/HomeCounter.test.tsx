import React from 'react'
import { render } from '../../test/testUtils'
import HomeCounter from './HomeCounter'

const Component = <HomeCounter />

describe('Counter', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(Component, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
