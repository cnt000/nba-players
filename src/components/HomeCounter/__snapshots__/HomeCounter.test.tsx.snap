// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdDrOH dJQgrP"
  >
    <header
      class="sc-dkXqmJ gnCqVg"
    >
      <img
        alt="logo"
        class="sc-gtczok hodEzD"
        src="logo.svg"
      />
      <div>
        <div
          class="row"
        >
          <button
            aria-label="Increment value"
            class="button"
          >
            +
          </button>
          <span
            class="value"
          >
            0
          </span>
          <button
            aria-label="Decrement value"
            class="button"
          >
            -
          </button>
        </div>
        <div
          class="row"
        >
          <input
            aria-label="Set increment amount"
            class="textbox"
            value="2"
          />
          <button
            class="button"
          >
            Add Amount
          </button>
          <button
            class="asyncButton"
          >
            Add Async
          </button>
        </div>
      </div>
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Hello World
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
